tests/gold_tests/continuations/openclose_h2.test.py

```diff
@@ -0,0 +1,143 @@
+    'proxy.config.ssl.server.cipher_suite': 'ECDHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA384:AES128-GCM-SHA256:AES256-GCM-SHA384:ECDHE-RSA-RC4-SHA:ECDHE-RSA-AES128-SHA:ECDHE-RSA-AES256-SHA:RC4-SHA:RC4-MD5:AES128-SHA:AES256-SHA:DES-CBC3-SHA!SRP:!DSS:!PSK:!aNULL:!eNULL:!SSLv2',
+    'proxy.config.http2.max_concurrent_streams_in': 65535
+})
+
+# add plugin to assist with test metrics
+Test.PreparePlugin(os.path.join(Test.Variables.AtsTestToolsDir,
+                                'plugins', 'ssntxnorder_verify.cc'), ts)
+
+comparator_command = '''
+if test "`traffic_ctl metric get ssntxnorder_verify.{0}.start | cut -d ' ' -f 2`" -eq "`traffic_ctl metric get ssntxnorder_verify.{0}.close | cut -d ' ' -f 2`" ; then\
+     echo yes;\
+    else \
+    echo no; \
+    fi;
+    '''
+
+numberOfRequests = 10
+
+tr = Test.AddTestRun()
+# curl with http2
+tr.Processes.Default.Command = 'for ((i=0; i<{0}; i++)); do curl --http2 -k https://127.0.0.1:{1}; done; sleep 5'.format(numberOfRequests, ts.Variables.ssl_port)
+tr.Processes.Default.ReturnCode = 0
+# time delay as proxy.config.http.wait_for_cache could be broken
+tr.Processes.Default.StartBefore(server, ready=When.PortOpen(server.Variables.Port))
+tr.Processes.Default.StartBefore(ts, ready=When.PortOpen(ts.Variables.ssl_port))
+tr.StillRunningAfter = ts
+
+# Watch the records snapshot file.
+records = ts.Disk.File(os.path.join(ts.Variables.RUNTIMEDIR, "records.snap"))
+
+# Check our work on traffic_ctl
+# no errors happened,
+tr = Test.AddTestRun()
+tr.DelayStart = 10
+tr.Processes.Default.Command = 'traffic_ctl metric get ssntxnorder_verify.err'
+tr.Processes.Default.ReturnCode = 0
+tr.Processes.Default.Env = ts.Env
+tr.Processes.Default.Streams.All = Testers.ContainsExpression(
+    'ssntxnorder_verify.err 0', 'incorrect statistic return, or possible error.')
+tr.StillRunningAfter = ts
+tr.StillRunningAfter = server
+
+# number of sessions/transactions opened and closed are equal
+tr = Test.AddTestRun()
+tr.Processes.Default.Command = comparator_command.format('ssn')
+tr.Processes.Default.ReturnCode = 0
+tr.Processes.Default.Env = ts.Env
+tr.Processes.Default.Streams.stdout = Testers.ContainsExpression(
+    "yes", 'should verify contents')
+tr.StillRunningAfter = ts
+tr.StillRunningAfter = server
+
+tr = Test.AddTestRun()
+tr.Processes.Default.Command = comparator_command.format('txn')
+tr.Processes.Default.ReturnCode = 0
+tr.Processes.Default.Env = ts.Env
+tr.Processes.Default.Streams.stdout = Testers.ContainsExpression(
+    "yes", 'should verify contents')
+tr.StillRunningAfter = ts
+tr.StillRunningAfter = server
+
+# session count is positive,
+tr = Test.AddTestRun()
+tr.Processes.Default.Command = "traffic_ctl metric get ssntxnorder_verify.ssn.start"
+tr.Processes.Default.ReturnCode = 0
+tr.Processes.Default.Env = ts.Env
+tr.Processes.Default.Streams.stdout = Testers.ExcludesExpression(
+    " 0", 'should be nonzero')
```

Comment:
This is testing that the output is `0`, but if it is not then we fail the test saying it should be nonzero?

Reply:
This is to see how many sessions have been closed. It should be non zero. If it is 0 this is an error. Note, it's `ExcludesExpression`.

Reply:
Ah, you're right. I misread it as another `ContainsExpression`.